modules/pubmaticBidAdapter.js

```diff
@@ -498,38 +605,86 @@ export const spec = {
         respCur = response.body.cur || respCur;
         response.body.seatbid.forEach(seatbidder => {
           seatbidder.bid &&
-          utils.isArray(seatbidder.bid) &&
-          seatbidder.bid.forEach(bid => {
-            let newBid = {
-              requestId: bid.impid,
-              cpm: (parseFloat(bid.price) || 0).toFixed(2),
-              width: bid.w,
-              height: bid.h,
-              creativeId: bid.crid || bid.id,
-              dealId: bid.dealid,
-              currency: respCur,
-              netRevenue: NET_REVENUE,
-              ttl: 300,
-              referrer: utils.getTopWindowUrl(),
-              ad: bid.adm
-            };
-            let parsedRequest = JSON.parse(request.data);
-            if (parsedRequest.imp && parsedRequest.imp.length > 0) {
-              parsedRequest.imp.forEach(req => {
-                if (bid.impid === req.id && req.hasOwnProperty('video')) {
-                  newBid.mediaType = 'video';
-                  newBid.width = bid.hasOwnProperty('w') ? bid.w : req.video.w;
-                  newBid.height = bid.hasOwnProperty('h') ? bid.h : req.video.h;
-                  newBid.vastXml = bid.adm;
-                }
-              });
-            }
-            if (bid.ext && bid.ext.deal_channel) {
-              newBid['dealChannel'] = dealChannelValues[bid.ext.deal_channel] || null;
-            }
-
-            bidResponses.push(newBid);
-          });
+            utils.isArray(seatbidder.bid) &&
+            seatbidder.bid.forEach(bid => {
+              let newBid = {
+                requestId: bid.impid,
+                cpm: (parseFloat(bid.price) || 0).toFixed(2),
+                width: bid.w,
+                height: bid.h,
+                creativeId: bid.crid || bid.id,
+                dealId: bid.dealid,
+                currency: respCur,
+                netRevenue: NET_REVENUE,
+                ttl: 300,
+                referrer: utils.getTopWindowUrl(),
+                ad: bid.adm
+              };
+              let parsedRequest = JSON.parse(request.data);
+              if (parsedRequest.imp && parsedRequest.imp.length > 0) {
+                parsedRequest.imp.forEach(req => {
+                  if (bid.impid === req.id && req.hasOwnProperty('video')) {
+                    newBid.mediaType = 'video';
+                    newBid.width = bid.hasOwnProperty('w') ? bid.w : req.video.w;
+                    newBid.height = bid.hasOwnProperty('h') ? bid.h : req.video.h;
+                    newBid.vastXml = bid.adm;
+                  }
+                  if (bid.impid === req.id && req.hasOwnProperty('native')) {
```

Comment:
move the native response handling to a function as the code is more than 5/10 lines

Reply:
Done